website/blog/2022-11-22-move-spreadsheets-to-your-dwh.md

```diff
@@ -0,0 +1,191 @@
+---
+title: "How to move data from spreadsheets into your data warehouse"
+description: "A thankless, humble, and inevitable task: getting spreadsheet data into your data warehouse. In this Developer Blog post, Joel walks through the different options for getting spreadsheet data into your data warehouse, and the pros and cons of each option."
+slug: moving-spreadsheet-data
+
+authors: [joel_labes]
+
+tags: [analytics craft]
+hide_table_of_contents: false
+
+date: 2022-11-23
+is_featured: true
+---
+
+Once your <Term id="data-warehouse"/> is built out, the vast majority of your data will have come from other SaaS tools, internal databases, or customer data platforms (CDPs). But there’s another unsung hero of the analytics engineering toolkit: the humble spreadsheet.
+
+Spreadsheets are the Swiss army knife of data processing. They can add extra context to otherwise inscrutable application identifiers, be the only source of truth for bespoke processes from other divisions of the business, or act as the translation layer between two otherwise incompatible tools.
+
+Because of spreadsheets’ importance as the glue between many business processes, there are different tools to load them into your data warehouse and each one has its own pros and cons, depending on your specific use case.
+
+<!--truncate-->
+
+In general, there are a few questions to ask yourself about your data before choosing one of these tools:
+
+- Who at your company will be loading the data?
+- Does it have a consistent format?
+- How frequently will it change?
+- How big is the dataset?
+- Do changes need to be tracked?
+- Where are the files coming from?
+
+Let’s have a look at some of the offerings to help you get your spreadsheets into your data warehouse.
+
+## dbt seeds
+
+If you’re using [dbt](https://www.getdbt.com/) to manage your data transformations, it comes with an inbuilt csv loader ([seeds](https://docs.getdbt.com/docs/building-a-dbt-project/seeds)) to populate your data warehouse with any files you put inside of your project’s `seeds` folder. It will automatically infer data types from your file’s contents, but you can always override it by [providing explicit instructions in your dbt_project.yml](https://docs.getdbt.com/reference/resource-configs/column_types) file.
+
+However, since dbt creates these tables by inserting rows one at a time, it doesn’t perform well at scale (there’s no hard limit but syncing a hundred rows should be fine, as opposed to a few thousand). [The dbt docs](https://docs.getdbt.com/docs/building-a-dbt-project/seeds#faqs) suggest using seeds for “files that contain business-specific logic, for example, a list of country codes or user IDs of employees.”
```

Comment:
```suggestion
However, since dbt creates these tables by inserting rows one at a time, it doesn’t perform well at scale (there’s no hard limit but aim for hundreds of rows rather than thousands). [The dbt docs](https://docs.getdbt.com/docs/building-a-dbt-project/seeds#faqs) suggest using seeds for “files that contain business-specific logic, for example, a list of country codes or user IDs of employees.”
```